ibis/backends/sql/compilers/pyspark.py

```diff
@@ -134,6 +135,11 @@ def visit_Field(self, op, *, rel, name):
             return result
 
     def visit_Cast(self, op, *, arg, to):
+        if isinstance(arg, sge.Func) and arg.name.upper() == "FROM_JSON":
+            return self.f.from_json(
+                arg.expressions[0],
+                sge.Literal.string(PySparkTypeMapper.from_ibis(to).simpleString()),
+            )
```

Comment:
I think a less fragile way to do this would be to instead check if `op.arg.type().is_json()` (or json array or json map) and then use `from_json` in that case. Checking for a specific function seems very fragile given that there are plenty of APIs that can produce JSON that aren't `.array` and `.map`.

```suggestion
        dtype = op.arg.dtype
        # does ibis think it's JSON, ARRAY<JSON> or MAP<STRING, JSON>?
        if dtype.is_json() or (dtype.is_array() and dtype.value_type.is_json()) or (dtype.is_map() and dtype.value_type.is_json()):
            return self.f.from_json(
                arg,
                PySparkTypeMapper.from_ibis(to).simpleString(),
            )
```

Reply:
This patch won't work. because this calls the `FROM_JSON` twice.
```python
t = ibis.literal('[{},{}]]').cast('json').name('a').as_table()
t1 = t.a.unwrap_as('array<json>')
t2 = t.a.unwrap_as('array<json>').cast('array<struct<>>')

backend.compile(t2)
```

output:
```sql
SELECT FROM_JSON(FROM_JSON(`t0`.`a`, 'array<string>'), 'array<struct<>>') AS `Cast(ToJSONArray(a), array<struct<>>)` FROM (SELECT '[{},{}]]' AS `a`) AS `t0`
```

we need to extract the 1th parameter from `from_json` function. ><